concordium-std/src/impls.rs

```diff
@@ -1640,6 +1640,24 @@ impl HasPolicy for Policy<AttributesCursor> {
         self.items.remaining_items -= 1;
         Some((AttributeTag(tag_value_len[0]), tag_value_len[1]))
     }
+
+    fn iter(&mut self) -> PolicyAttributesIter<Self> {
```

Comment:
Calling next on the iterator will mutate the implementor of `HasPolicy`, which is not ideal.
It would be nice, if we could avoid the need for a mutable reference to self here.

Ideally, we could also remove the `next_item` in `HasPolicy`, but is a breaking change of course.